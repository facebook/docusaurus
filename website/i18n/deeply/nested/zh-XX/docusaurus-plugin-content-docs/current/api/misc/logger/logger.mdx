---
sidebar_position: 2
slug: /api/misc/@docusaurus/logger
---

# 📦 logger

一个封装好的记录器，用于输出根据语义格式化的控制台信息。

我们鼓励 Docusaurus 生态系统中的包作者用这个包提供统一的日志格式。

## API

它导出单个对象作为默认导出：`logger`。 `logger` 具有如下属性：

- 一些有用的颜色。
  - `红色`
  - `黄色`
  - `绿色`
  - `粗体`
  - `调暗`
- 格式化器。 所有这些函数都有形如 `(msg: unknown) => string` 的签名。 请注意，我们不对它们的具体实现作保证。 你只应关心它们的语义。
  - `path`: 格式化文件路径。
  - `url`: 格式化 URL。
  - `name`: 格式化标识符。
  - `code`: 格式化小代码片段。
  - `subdue`: 让文本变得不显眼。
  - `num`: 格式化数字。
- `interpolate` 函数。 这是一个模板字符串标签。 具体语法请见后文。
- 输出函数。 所有的输出函数都既可以当作正常函数调用（类似于 `console.log`，但只接受一个参数），也可以作为模板字符串标签。
  - `info`: 打印信息。
  - `warn`: 打印一个应引起注意的警告。
  - `error`: 打印一个错误（不一定会停止程序），表明存在重大问题。
  - `success`: 打印成功消息。
- `report` 函数。 它需要 `ReportingSeverity` 值(`value `, `log`, `warn`, `throw`) 并根据严重程度报告消息。

:::warning 关于 `error` 格式化器的说明

请注意，即使程序没有中断，`error` 信息也很可能让人迷惑。 当用户检查日志，并找到一行 `[ERROR]` 时，即使构建成功了，他们也会假定出现了错误。 尽量少使用它。

Docusaurus 只会在抛出错误之前用 `logger.error` 打印信息，或者当用户把 `onBrokenLink` 等选项设置成 `"error"` 时。

此外， `warn` 和 `error` 会给**整个**信息添加颜色，以吸引人的注意。 如果你正在打印关于某个错误的大块帮助提示，最好用 `logger.info`。

:::

### 使用模板字符串标签

模板字符串标签会解析模板和其中嵌入的表达式。 `interpolate` 会返回一个新字符串，而其他日志函数则会把结果打印出来。 以下是一个典型用例：

```js
import logger from '@docusaurus/logger';

logger.info`Hello name=${name}! You have number=${money} dollars. Here are the ${
  items.length > 1 ? 'items' : 'item'
} on the shelf: ${items}
To buy anything, enter code=${'buy x'} where code=${'x'} is the item's name; to quit, press code=${'Ctrl + C'}.`;
```

嵌入的表达式可以在前面包含一个标志，形式是 `[a-z]+=`（若干小写字母，随后是一个等于符号）。 如果表达式前面没有任何标志，它会按原样打印。 否则，它会被用下列的格式化器之一格式化：

- `path=`: `path`
- `url=`: `url`
- `name=`: `name`
- `code=`: `code`
- `subdue=`: `subdue`
- `number=`: `num`

如果表达式是一个数组，它会被以 `` "\n- ${array.join('\n- ')}\n` `` 的形式被格式化（注意它会自动有一个开头的换行）。 列表的每个元素都会被独立格式化，开头的横杠不会被格式化。 所以上面的消息会被打印成：

![一些文本在终端中输出，包含数组、代码、名称和数字格式](./demo.png)
