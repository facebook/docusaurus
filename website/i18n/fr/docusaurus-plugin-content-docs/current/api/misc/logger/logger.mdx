---
sidebar_position: 2
slug: /api/misc/@docusaurus/logger
---

# üì¶ logger

Un g√©n√©rateur de journaux encapsul√© pour le formatage s√©mantique des messages de la console.

Les auteurs de paquets de l'√©cosyst√®me Docusaurus sont encourag√©s √† utiliser ce paquet pour fournir des formats de log unifi√©s.

## API

Il exporte un seul objet comme exportation par d√©faut¬†: `logger`. `logger` a les propri√©t√©s suivantes¬†:

- Quelques couleurs utiles.
  - `red`
  - `yellow`
  - `green`
  - `bold`
  - `dim`
- Formatteurs. Ces fonctions ont toutes la signature `(msg: unknown) => string`. Notez que leurs impl√©mentations ne sont pas garanties. Vous ne devriez vous pr√©occuper que de leur s√©mantique.
  - `path`¬†: formate un chemin de fichier.
  - `url`¬†: formate une URL.
  - `name`¬†: formate un identifiant.
  - `code`¬†: formate un extrait de code.
  - `subdue`¬†: soumet le texte.
  - `num`¬†: formate un nombre.
- La fonction `interpolate`. C'est un gabarit √©tiquet√©. La syntaxe peut √™tre trouv√©e ci-dessous.
- Fonctions de journalisation. Toutes les fonctions de journalisation peuvent √† la fois √™tre utilis√©es comme des fonctions normales (similaires √† la famille `console.log`, mais n'acceptant qu'un seul param√®tre) ou des gabarits √©tiquet√©s.
  - `info`¬†: affiche une information.
  - `warn`¬†: affiche un avertissement auquel il faut pr√™ter attention.
  - `error`¬†: affiche une erreur (sans n√©cessairement arr√™ter le programme) qui signale des probl√®mes importants.
  - `success`¬†: affiche un message de r√©ussite.
- La fonction `report`. Il prend une valeur de `ReportingSeverity` (`ignore`, `log`, `warn`, `throw`) et signale un message en fonction de la gravit√©.

:::warning Un mot sur le formateur `error`

Attention, un message `error`, m√™me s'il ne bloque pas le programme, risque de cr√©er la confusion. Lorsque les utilisateurs inspectent les logs et trouvent un `[ERROR]`, m√™me lorsque la construction r√©ussit, ils supposent que quelque chose ne va pas. Utilisez-le avec mod√©ration.

Docusaurus n'utilise `logger.error` que lorsqu'il s'agit d'afficher des messages imm√©diatement avant de lancer une erreur, ou lorsque l'utilisateur a d√©fini la gravit√© du rapport de `onBrokenLink`, etc. sur `"error"`.

En outre, `warn` et `error` coloriseront **l'ensemble** du message pour attirer l'attention. Si vous affichez de gros blocs d'aide √† propos d'une erreur, mieux vaut utiliser `logger.info`.

:::

### Utilisation du gabarit √©tiquet√©

Le gabarit √©tiquet√© √©value le template et les expressions int√©gr√©es. `interpolate` retourne une nouvelle cha√Æne, tandis que d'autres fonctions de journalisation l'affichent. Ci-dessous est une utilisation typique¬†:

```js
import logger from '@docusaurus/logger';

logger.info`Hello name=${name}! Vous avez number=${money} dollars. Voici ${
  items.length > 1 ? 'les articles' : 'l'\article'
} de l'√©tag√®re¬†: ${items}
Pour acheter quelque chose, saisissez code=${'acheter x'} o√π code=${'x'} est le nom de l'article; pour quitter, pressez code=${'Ctrl + C'}.`;
```

Une expression int√©gr√©e est √©ventuellement pr√©c√©d√©e d'un drapeau de la forme `[a-z]+=` (quelques minuscules, suivi d'un signe √©gal qui pr√©c√®de directement l'expression int√©gr√©e). Si l'expression n'est pr√©c√©d√©e d'aucun drapeau, elle est imprim√©e telle quelle. Sinon, il est format√© avec un des formateurs¬†:

- `path=`: `path`
- `url=`: `url`
- `name=`: `name`
- `code=`: `code`
- `subdue=`: `subdue`
- `number=`: `num`

Si l'expression est un tableau, elle est format√©e par `` `\n- ${array.join('\n- ')}\n` `` (notez qu'elle obtient automatiquement un saut de ligne √† la fin). Chaque membre est format√© par lui-m√™me et la puce n'est pas format√©e. Vous verrez donc le message ci-dessus affich√© sous la forme suivante :

![Une sortie de texte dans le terminal, contenant le tableau, le code, le nom et le formatage des chiffres](./demo.png)
