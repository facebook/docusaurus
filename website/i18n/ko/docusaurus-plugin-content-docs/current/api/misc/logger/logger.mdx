---
sidebar_position: 2
slug: /api/misc/@docusaurus/logger
---

# 로거

의미있는 포맷으로 콘솔 메시지를 지정하기 위해 캡슐화된 로거입니다.

도큐사우루스 생태계 내에서 패키지를 작성한다면 이 패키지를 사용해 통합 로그 형식을 제공하는 것을 권장합니다.

## API

기본 export 설정은 단일 오브젝트 `logger` 입니다. `logger`에는 다음의 속성이 있습니다.

- 일부 유용한 색상
  - `red`
  - `yellow`
  - `green`
  - `bold`
  - `dim`
- 포맷터. 포맷터 함수는 모두 `(msg: unknown) => string` 시그니처를 가지고 있습니다. 기능 구현이 보장되는 건 아니라는 점에 유의하세요. 각각의 의미만 참고하세요.
  - `path`: 파일 경로 포맷.
  - `url`: URL 포맷.
  - `name`: 식별자 포맷.
  - `code`: 코드 스니핏 포맷.
  - `subdue`: 텍스트를 덜 눈에 띄게 설정.
  - `num`: 숫자 포맷.
- `interpolate` 함수. 템플릿 리터럴 태그입니다. 문법은 아래 로깅 함수와 비슷합니다.
- 로깅 함수 모든 로깅 함수는 일반적인 함수(`console.log`와 비슷하지만 하나의 파라미터만 허용합니다) 또는 템플릿 리터널 태그로 사용할 수있습니다.
  - `info`: 정보성 데이터를 남깁니다.
  - `warn`: 주의가 필요한 경고성 데이터를 남깁니다.
  - `error`: 심각한 문제를 알리는 오류(프로그램을 중지할 필요는 없는) 정보를 남깁니다.
  - `success`: 성공 관련 메시지를 남깁니다.
- `report` 함수. `ReportingSeverity` 값(`ignore`, `log`, `warn`, `throw`)을 받아 심각도에 따라 메시지로 알려줍니다.

:::warning A word on the `error` formatter

`error` 메시지는 프로그램을 중단시키지 않지만 혼동을 일으킬 수 있으므로 주의해야 합니다. 사용자가 로그를 조사하고 `[ERROR]`를 발견하면 빌드가 성공했더라도 뭔가 잘못되었다고 가정할 수 있습니다. 신중하게 사용하세요.

도큐사우루스는 오류가 발생하기 직전에 메시지를 남기거나 사용자가 `onBrokenLink` 등의 리포트 심각도를 `"error"`로 설정한 경우에만 `logger.error`를 사용합니다.

또한 `warn`이나 `error`는 사용자의 주의를 끌기 위해 **전체** 메시지에 색상을 적용합니다. 오류에 대한 도움말 텍스트 같은 많은 메시지를 남기는 경우에는 `logger.info`를 사용하는 것이 좋습니다.

:::

### 템플릿 리터럴 태그 사용하기

템플릿 리터럴 태그는 템플릿과 포함된 평가식을 처리합니다. `interpolate`는 다른 로깅 함수가 메시지를 남기는 것과 다르게 새로운 문자열을 반환합니다. 다음은 일반적인 사용법입니다.

```js
import logger from '@docusaurus/logger';

logger.info`Hello name=${name}! You have number=${money} dollars. Here are the ${
  items.length > 1 ? 'items' : 'item'
} on the shelf: ${items}
To buy anything, enter code=${'buy x'} where code=${'x'} is the item's name; to quit, press code=${'Ctrl + C'}.`;
```

포함된 표현식은 `[a-z]+=` 형식(포함된 표현식 바로 앞에 몇 개의 소문자, 등호가 따라옵니다)의 플래그가 선택적으로 앞에 옵니다. 표현식 앞에 플래그가 없으면 그대로 메시지를 남깁니다. 그렇지 않으면 포맷터 중 하나로 포맷이 설정됩니다.

- `path=`: `path`
- `url=`: `url`
- `name=`: `name`
- `code=`: `code`
- `subdue=`: `subdue`
- `number=`: `num`

표현식이 배열인 경우에는 `` `\n- ${array.join('\n- ')}\n` `` 형식으로 처리됩니다(자동으로 앞 줄 끝에 이어서 메시지를 남깁니다). 각 항목은 자체적으로 포맷이 처리되며 글머리 기호는 포맷이 지정되지 않습니다. 따라서 위의 메시지는 다음과 같이 처리됩니다.

![터미널에 표시된 배열, 코드, 이름, 숫자 형식을 포함하는 텍스트](./demo.png)
