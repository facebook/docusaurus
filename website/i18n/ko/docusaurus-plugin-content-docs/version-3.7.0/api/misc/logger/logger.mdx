---
sidebar_position: 2
slug: /api/misc/@docusaurus/logger
---

# 로거

의미있는 포맷으로 콘솔 메시지를 지정하기 위해 캡슐화된 로거입니다.

도큐사우루스 생태계 내에서 패키지를 작성한다면 이 패키지를 사용해 통합 로그 형식을 제공하는 것을 권장합니다.

## API

It exports a single object as default export: `logger`. `logger` has the following properties:

- 일부 유용한 색상
  - `red`
  - `yellow`
  - `green`
  - `bold`
  - `dim`
- 포맷터. These functions all have the signature `(msg: unknown) => string`. 기능 구현이 보장되는 건 아니라는 점에 유의하세요. 각각의 의미만 참고하세요.
  - `path`: formats a file path.
  - `url`: formats a URL.
  - `name`: formats an identifier.
  - `code`: formats a code snippet.
  - `subdue`: subdues the text.
  - `num`: formats a number.
- The `interpolate` function. 템플릿 리터럴 태그입니다. 문법은 아래 로깅 함수와 비슷합니다.
- 로깅 함수 All logging functions can both be used as normal functions (similar to the `console.log` family, but only accepts one parameter) or template literal tags.
  - `info`: prints information.
  - `warn`: prints a warning that should be paid attention to.
  - `error`: prints an error (not necessarily halting the program) that signals significant problems.
  - `success`: prints a success message.
- The `report` function. It takes a `ReportingSeverity` value (`ignore`, `log`, `warn`, `throw`) and reports a message according to the severity.

:::warning A word on the `error` formatter

Beware that an `error` message, even when it doesn't hang the program, is likely going to cause confusion. When users inspect logs and find an `[ERROR]`, even when the build succeeds, they will assume something is going wrong. 신중하게 사용하세요.

Docusaurus only uses `logger.error` when printing messages immediately before throwing an error, or when user has set the reporting severity of `onBrokenLink`, etc. to `"error"`.

In addition, `warn` and `error` will color the **entire** message for better attention. If you are printing large blocks of help text about an error, better use `logger.info`.

:::

### 템플릿 리터럴 태그 사용하기

템플릿 리터럴 태그는 템플릿과 포함된 평가식을 처리합니다. `interpolate` returns a new string, while other logging functions prints it. 다음은 일반적인 사용법입니다.

```js
import logger from '@docusaurus/logger';

logger.info`Hello name=${name}! You have number=${money} dollars. Here are the ${
  items.length > 1 ? 'items' : 'item'
} on the shelf: ${items}
To buy anything, enter code=${'buy x'} where code=${'x'} is the item's name; to quit, press code=${'Ctrl + C'}.`;
```

An embedded expression is optionally preceded by a flag in the form `[a-z]+=` (a few lowercase letters, followed by an equals sign, directly preceding the embedded expression). 표현식 앞에 플래그가 없으면 그대로 메시지를 남깁니다. 그렇지 않으면 포맷터 중 하나로 포맷이 설정됩니다.

- `path=`: `path`
- `url=`: `url`
- `name=`: `name`
- `code=`: `code`
- `subdue=`: `subdue`
- `number=`: `num`

If the expression is an array, it's formatted by `` `\n- ${array.join('\n- ')}\n` `` (note it automatically gets a leading line end). 각 항목은 자체적으로 포맷이 처리되며 글머리 기호는 포맷이 지정되지 않습니다. 따라서 위의 메시지는 다음과 같이 처리됩니다.

![Some text output in the terminal, containing array, code, name, and number formatting](./demo.png)
