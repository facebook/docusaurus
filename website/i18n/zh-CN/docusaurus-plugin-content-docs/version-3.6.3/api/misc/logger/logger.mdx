---
sidebar_position: 2
slug: /api/misc/@docusaurus/logger
---

# 📦 logger

一个封装好的记录器，用于输出根据语义格式化的控制台信息。

我们鼓励 Docusaurus 生态系统中的包作者用这个包提供统一的日志格式。

## API

It exports a single object as default export: `logger`. `logger` has the following properties:

- 一些有用的颜色。
  - `red`
  - `yellow`
  - `green`
  - `bold`
  - `dim`
- 格式化器。 These functions all have the signature `(msg: unknown) => string`. 请注意，我们不对它们的具体实现作保证。 你只应关心它们的语义。
  - `path`: formats a file path.
  - `url`: formats a URL.
  - `name`: formats an identifier.
  - `code`: formats a code snippet.
  - `subdue`: subdues the text.
  - `num`: formats a number.
- The `interpolate` function. 这是一个模板字符串标签。 具体语法请见后文。
- 输出函数。 All logging functions can both be used as normal functions (similar to the `console.log` family, but only accepts one parameter) or template literal tags.
  - `info`: prints information.
  - `warn`: prints a warning that should be paid attention to.
  - `error`: prints an error (not necessarily halting the program) that signals significant problems.
  - `success`: prints a success message.
- The `report` function. It takes a `ReportingSeverity` value (`ignore`, `log`, `warn`, `throw`) and reports a message according to the severity.

:::warning A word on the `error` formatter

Beware that an `error` message, even when it doesn't hang the program, is likely going to cause confusion. When users inspect logs and find an `[ERROR]`, even when the build succeeds, they will assume something is going wrong. 尽量少使用它。

Docusaurus only uses `logger.error` when printing messages immediately before throwing an error, or when user has set the reporting severity of `onBrokenLink`, etc. to `"error"`.

In addition, `warn` and `error` will color the **entire** message for better attention. If you are printing large blocks of help text about an error, better use `logger.info`.

:::

### 使用模板字符串标签

模板字符串标签会解析模板和其中嵌入的表达式。 `interpolate` returns a new string, while other logging functions prints it. 以下是一个典型用例：

```js
import logger from '@docusaurus/logger';

logger.info`Hello name=${name}! You have number=${money} dollars. Here are the ${
  items.length > 1 ? 'items' : 'item'
} on the shelf: ${items}
To buy anything, enter code=${'buy x'} where code=${'x'} is the item's name; to quit, press code=${'Ctrl + C'}.`;
```

An embedded expression is optionally preceded by a flag in the form `[a-z]+=` (a few lowercase letters, followed by an equals sign, directly preceding the embedded expression). 如果表达式前面没有任何标志，它会按原样打印。 否则，它会被用下列的格式化器之一格式化：

- `path=`: `path`
- `url=`: `url`
- `name=`: `name`
- `code=`: `code`
- `subdue=`: `subdue`
- `number=`: `num`

If the expression is an array, it's formatted by `` `\n- ${array.join('\n- ')}\n` `` (note it automatically gets a leading line end). 列表的每个元素都会被独立格式化，开头的横杠不会被格式化。 所以上面的消息会被打印成：

![Some text output in the terminal, containing array, code, name, and number formatting](./demo.png)
