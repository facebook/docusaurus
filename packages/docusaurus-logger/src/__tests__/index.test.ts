/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import {jest} from '@jest/globals';
import logger from '../index';

describe('formatters', () => {
  it('path', () => {
    expect(logger.path('keepAnsi')).toMatchInlineSnapshot(
      `"<cyan><underline>"keepAnsi"</underline></color>"`,
    );
  });
  it('url', () => {
    expect(logger.url('https://docusaurus.io/keepAnsi')).toMatchInlineSnapshot(
      `"<cyan><underline>https://docusaurus.io/keepAnsi</underline></color>"`,
    );
  });
  it('id', () => {
    expect(logger.name('keepAnsi')).toMatchInlineSnapshot(
      `"<blue><bold>keepAnsi</intensity></color>"`,
    );
  });
  it('code', () => {
    expect(logger.code('keepAnsi')).toMatchInlineSnapshot(
      `"<cyan>\`keepAnsi\`</color>"`,
    );
  });
  it('subdue', () => {
    expect(logger.subdue('keepAnsi')).toMatchInlineSnapshot(
      `"<gray>keepAnsi</color>"`,
    );
  });
});

describe('interpolate', () => {
  it('formats text with variables & arrays', () => {
    const name = 'Josh';
    const items = [1, 'hi', 'Hmmm'];
    expect(
      logger.interpolate`(keepAnsi) Hello ${name}! Here are your goodies:${items}`,
    ).toMatchInlineSnapshot(`
      "(keepAnsi) Hello Josh! Here are your goodies:
      - 1
      - hi
      - Hmmm"
    `);
  });
  it('recognizes valid flags', () => {
    expect(
      logger.interpolate`(keepAnsi) The package at path=${'packages/docusaurus'} has number=${10} files. name=${'Babel'} is exported here subdue=${'(as a preset)'} that you can with code=${"require.resolve('@docusaurus/core/lib/babel/preset')"}`,
    ).toMatchInlineSnapshot(
      `"(keepAnsi) The package at <cyan><underline>"packages/docusaurus"</underline></color> has <yellow>10</color> files. <blue><bold>Babel</intensity></color> is exported here <gray>(as a preset)</color> that you can with <cyan>\`require.resolve('@docusaurus/core/lib/babel/preset')\`</color>"`,
    );
  });
  it('interpolates arrays with flags', () => {
    expect(
      logger.interpolate`(keepAnsi) The following commands are available:code=${[
        'docusaurus start',
        'docusaurus build',
        'docusaurus deploy',
      ]}`,
    ).toMatchInlineSnapshot(`
      "(keepAnsi) The following commands are available:
      - <cyan>\`docusaurus start\`</color>
      - <cyan>\`docusaurus build\`</color>
      - <cyan>\`docusaurus deploy\`</color>"
    `);
  });
  it('prints detached flags as-is', () => {
    expect(
      logger.interpolate`(keepAnsi) You can use placeholders like code= ${'and it will'} be replaced with the succeeding arguments`,
    ).toMatchInlineSnapshot(
      `"(keepAnsi) You can use placeholders like code= and it will be replaced with the succeeding arguments"`,
    );
  });
  it('throws with bad flags', () => {
    expect(
      () =>
        logger.interpolate`(keepAnsi) I mistyped this: cde=${'this code'} and I will be damned`,
    ).toThrowErrorMatchingInlineSnapshot(
      `"Bad Docusaurus logging message. This is likely an internal bug, please report it."`,
    );
  });
});

describe('info', () => {
  const consoleMock = jest.spyOn(console, 'info').mockImplementation(() => {});
  it('prints objects', () => {
    logger.info({a: 1});
    logger.info(undefined);
    logger.info([1, 2, 3]);
    logger.info(new Date(2021, 10, 13));
    expect(consoleMock.mock.calls).toMatchSnapshot();
  });
});

describe('warn', () => {
  const consoleMock = jest.spyOn(console, 'warn').mockImplementation(() => {});
  it('prints objects', () => {
    logger.warn({a: 1});
    logger.warn(undefined);
    logger.warn([1, 2, 3]);
    logger.warn(new Date(2021, 10, 13));
    expect(consoleMock.mock.calls).toMatchSnapshot();
  });
});

describe('error', () => {
  const consoleMock = jest.spyOn(console, 'error').mockImplementation(() => {});
  it('prints objects', () => {
    logger.error({a: 1});
    logger.error(undefined);
    logger.error([1, 2, 3]);
    logger.error(new Date(2021, 10, 13));
    expect(consoleMock.mock.calls).toMatchSnapshot();
  });
});

describe('success', () => {
  const consoleMock = jest.spyOn(console, 'log').mockImplementation(() => {});
  it('prints objects', () => {
    logger.success({a: 1});
    logger.success(undefined);
    logger.success([1, 2, 3]);
    logger.success(new Date(2021, 10, 13));
    expect(consoleMock.mock.calls).toMatchSnapshot();
  });
});

describe('report', () => {
  beforeAll(() => jest.clearAllMocks());
  it('works with all severities', () => {
    const consoleLog = jest.spyOn(console, 'info').mockImplementation(() => {});
    const consoleWarn = jest
      .spyOn(console, 'warn')
      .mockImplementation(() => {});
    logger.report('ignore')('hey');
    logger.report('log')('hey');
    logger.report('warn')('hey');
    expect(() =>
      logger.report('throw')('hey'),
    ).toThrowErrorMatchingInlineSnapshot(`"hey"`);
    expect(() =>
      // @ts-expect-error: for test
      logger.report('foo')('hey'),
    ).toThrowErrorMatchingInlineSnapshot(
      `"Unexpected "reportingSeverity" value: foo."`,
    );
    expect(consoleLog).toHaveBeenCalledTimes(1);
    expect(consoleLog).toHaveBeenCalledWith(
      expect.stringMatching(/.*\[INFO\].* hey/),
    );
    expect(consoleWarn).toHaveBeenCalledTimes(1);
    expect(consoleWarn).toHaveBeenCalledWith(
      expect.stringMatching(/.*\[WARNING\].* hey/),
    );
  });
});
