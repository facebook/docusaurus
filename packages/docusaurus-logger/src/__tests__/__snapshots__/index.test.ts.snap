// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error prints objects 1`] = `
[
  [
    "[ERROR] {"a":1}",
  ],
  [
    "[ERROR] undefined",
  ],
  [
    "[ERROR] 1,2,3",
  ],
  [
    "[ERROR] Sat, 13 Nov 2021 00:00:00 GMT",
  ],
]
`;

exports[`info prints objects 1`] = `
[
  [
    "[INFO] {"a":1}",
  ],
  [
    "[INFO] undefined",
  ],
  [
    "[INFO] 1,2,3",
  ],
  [
    "[INFO] Sat, 13 Nov 2021 00:00:00 GMT",
  ],
]
`;

exports[`success prints objects 1`] = `
[
  [
    "[SUCCESS] {"a":1}",
  ],
  [
    "[SUCCESS] undefined",
  ],
  [
    "[SUCCESS] 1,2,3",
  ],
  [
    "[SUCCESS] Sat, 13 Nov 2021 00:00:00 GMT",
  ],
]
`;

exports[`warn prints objects 1`] = `
[
  [
    "[WARNING] {"a":1}",
  ],
  [
    "[WARNING] undefined",
  ],
  [
    "[WARNING] 1,2,3",
  ],
  [
    "[WARNING] Sat, 13 Nov 2021 00:00:00 GMT",
  ],
]
`;
